backend/main-service/src/main/java/kkakka/mainservice/order/domain/Order.java

```diff
@@ -1,4 +1,77 @@
 package kkakka.mainservice.order.domain;
 
+import kkakka.mainservice.member.domain.Member;
+import lombok.Getter;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table(name = "`ORDER`")
+@Getter
 public class Order {
+
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "member_id")
+    private Member member; //주문회원
+
+    //TODO: 2022.09.28 배송지정보 추가할 것 -hyeyeon
+
+    private LocalDateTime orderedAt; //주문시간
+
+    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
+    private List<ProductOrder> productOrders = new ArrayList<>();
+    private Integer totalPrice; //총금액
+
+    //==연관관계 메서드==//
+    public void setMember(Member member) {
+        this.member = member;
+        member.getOrders().add(this); //회원에게 주문 추가
+    }
+
+    public void addProductOrder(ProductOrder productOrder) {
+        productOrders.add(productOrder);
+        productOrder.setOrder(this);
+    }
+
+    //==생성 메서드==//
+    public static Order createOrder(Member member, int totalPrice, List<ProductOrder> productOrders) {
+        Order order = new Order();
+        order.setMember(member);
+        order.setTotalPrice(totalPrice);
+
+        //상품주문 추가
+        for(ProductOrder productOrder : productOrders) {
+            order.addProductOrder(productOrder);
+        }
+
+        order.setOrderedAt(LocalDateTime.now());
+
+        return order;
```

Comment:
```suggestion
      new Order(null, member, LocalDateTime.now(), productOrders, totalPrice);
```

모든 필드를 매개변수로 가지는 주생성자를 만들고 위처럼 표현하면 더 좋을 것 같아요.
setter 를 꼭 써야하는 게 아니라면 최대한 쓰지 않고 싶어서요.

왜 객체지향 프로그래밍에서 setter 를 지양하는지 찾아보는거 추천!

Reply:
네! setter의 경우 객체의 일관성 때문이라고 알고 있는데 생성자를 사용하면 매개변수에 따른 대응이나 가독성에 문제도 있다고 하네요. 그 방안으로 빌더 패턴을 사용한다고 하는데 저희 상황에서도 좋은 방안인건지? 또 그만큼 많이 사용하는 편인지도 궁금해지네요.
~~(@builder를 사용하지않으면 클래스가 좀 지저분해 보이기도 하고,,,)~~
잘 몰라서 의견이 듣고싶습니다!

Reply:
일단 `@Builder` 는 롬복이 제공해주는 기능이라 다른 대안을 찾거나 직접 빌더를 구현하거나 하는 방법들이 있겠져.
(롬복은 코드가 라이브러리 의존적이 되고 사이드 이펙트가 발생할 수 있기 때문에 호불호가 있는 기술입니다.)
부생성자를 만들거나 `정적 팩토리 패턴`을 활용해서 생성할 수도 있습니다. 물론 이 방법들도 매개변수 타입이나 이런거 때문에 말한것처럼 대응에 문제가 생길 수도 있어서 그때는 빌더 패턴이 더 효율적일거 같네요~
https://kwonnam.pe.kr/wiki/java/lombok/pitfall
여기도 참고하면 좋을거 같아요!

Reply:
오오... 정적 팩토리 패턴... 
정말 많이 배워갑니다.☺ 참고 자료까지 감사합니다!